ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/recovery/PeriodicRecovery.java

```diff
@@ -859,6 +898,12 @@ private void restoreClassLoader(ClassLoader cl)
     private void loadModules ()
     {
         _recoveryModules.addAll(recoveryPropertyManager.getRecoveryEnvironmentBean().getRecoveryModules());
+
+        if (!_recoveryModules.isEmpty() &&
+                _recoveryModules.stream().anyMatch(
+                        x -> x.getClass().getName().equals("AtomicActionRecoveryModule"))) {
```

Comment:
I think the comparison that is intended here would be `com.arjuna.ats.internal.arjuna.recovery.AtomicActionRecoveryModule`?

Reply:
But I think we should be looking for any configured classes that are not `com.arjuna.ats.internal.arjuna.recovery.AtomicActionRecoveryModule`

Reply:
So `x -> !x.getClass().getName().equals("com.arjuna.ats.internal.arjuna.recovery.AtomicActionRecoveryModule"))) {`

Reply:
We also need similar check and potential alerting to be added in addModule after a module is manually added: https://github.com/jbosstm/narayana/blob/9da7020ce86d109fe997cb8006ff8c3d95a71a08/ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/recovery/PeriodicRecovery.java#L518

Reply:
fair

Reply:
The code has been modified. Thanks for spotting the bug, Tom :-)

Reply:
You are welcome, thank you